### PR TITLE
Allow scoping searches to site:

### PR DESCRIPTION
**Pull request by sywhb**

### CODE REVIEW

**Comment by sywhb** [comment]
#1907

**Comment by sywhb** [comment]
Because `url` is of `keyword` type in elasticsearch so we could not search for the domain and do a `match` query against it.
I'm thinking about adding a `domain` or `hostname` of `text` type in the `page` document in elasticsearch to make it searchable so we could do a `multi_match` against both `siteName` and `domain`

**Comment by sywhb** [comment]
> Because `url` is of `keyword` type in elasticsearch so we could not search for the domain (url is not tokenized) and do a `match` query against it.
> 
> I'm thinking about adding a `domain` or `hostname` of `text` type in the `page` document in elasticsearch to make it searchable so we could do a `multi_match` against both `siteName` and `domain`

Resolved this with `wildcard` query:
```
{
    should: [
      {
        match: {
          siteName,
        },
      },
      {
        wildcard: {
          // siteName is a domain name, so we need to wildcard the end
          url: `*${siteName}*`,
        },
      },
    ],
    minimum_should_match: 1,
  }```

**Comment by sywhb** [comment]
This is ready

**Comment by sywhb** [comment]
Fallback to hostname if site name is `null`

File: packages/api/.eslintignore [hunk 1]
```diff
@@ -3,3 +3,4 @@ dist/
 readabilityjs/
 src/generated/
 test/resolvers/
+apollo.config.js
```
Comment (sywhb) [inline packages/api/.eslintignore:6]: For Apollo extension to work in VSCode

Comment (jacksonh) [reply packages/api/.eslintignore:6]: nice. It only works in dev mode, right?

**Review by jacksonh — COMMENTED** [review]
One question on this: i saw the `siteName` added to tests, was this needed because something would fail if siteName is null?

**Review by jacksonh — APPROVED** [review]

File: packages/api/.eslintignore (hunk 1 above)
Comment (sywhb) [reply packages/api/.eslintignore:6]: yeah, but actually i don't really need to add this line because `apollo.config.js` is a Javascript file

**Comment by sywhb** [comment]
> One question on this: i saw the `siteName` added to tests, was this needed because something would fail if siteName is null?

No, i think it might be better to have the host or domain name as the default `siteName` and these tests would fail because their `siteName` are `null` from the previous generation

**Pull request merged**